src/peft/tuners/ia3/layer.py

```diff
@@ -294,7 +291,7 @@ def unmerge(self) -> None:
                     base_layer.bias.data = torch.mul(base_layer.bias.data, scaling.data)
 
     def forward(self, x: torch.Tensor, *args: Any, **kwargs: Any) -> torch.Tensor:
-        previous_dtype = x.dtype
+        dtype = previous_dtype = x.dtype
```

Comment:
`dtpye` was not defined but it is used in https://github.com/huggingface/peft/blob/f1ecfa6ae6eba599ae89decbf47b339d8c9d39a3/src/peft/tuners/ia3/layer.py#L314